src/main.ts

```diff
@@ -74,3 +72,36 @@ function isGhes(): boolean {
   );
   return ghUrl.hostname.toUpperCase() !== 'GITHUB.COM';
 }
+
+function resolveVersionInput(): string {
+  let version = core.getInput('node-version') || core.getInput('version');
+  const versionFileInput = core.getInput('node-version-file');
+
+  if (version && versionFileInput) {
+    core.warning(
+      'Both node-version and node-version-file inputs are specified, only node-version will be used'
+    );
+  }
+
+  if (version) {
+    return version;
+  }
+
+  if (versionFileInput) {
```

Comment:
We should have customers fall into the pit of success.  That includes the friction free path of (1) there's a version file in the root of the repo ( common) and (2) didnt specify a version input or version file.  In that common case it should just work. 

So I think there should be a case here version file exists and not version.  use it.

or maybe I'm reading all this wrong and that will work.

Reply:
> So I think there should be a case here version file exists and not version. use it.

Essentially we are doing that - just not explicitly saying it.  if `version-file` is specified  there would be no need to check for `version` since it would have been used and returned  before  checking  for `version-file`

Reply:
I think you still need to explicitly provide a `version-file` input though? 

I'm with @bryanmacfarlane. If you have added a common node version file to your repo you've _already_ expressed the version you want to use and how you want to express it, and it seems unnecessary to have to tell the action about it. 

This is how [travis does it](https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#specifying-nodejs-versions-using-nvmrc) for example (although only for .`nvmrc`).

It is slightly complicated by the number of options (the reason I used [preferred-node-version](https://www.npmjs.com/package/preferred-node-version) in #239) but what is the value to users in having to direct the action to a specific file?

Reply:
> > So I think there should be a case here version file exists and not version. use it.
> 
> Essentially we are doing that - just not explicitly saying it. if `version-file` is specified there would be no need to check for `version` since it would have been used and returned before checking for `version-file`

I agree with @Xlient , if there is `version-file` only  then it will also work simply with its inputs

Reply:
> I agree with @Xlient , if there is `version-file` only then it will also work simply with its inputs

agreed – it will totally work, but why is it _required_?

what is the benefit of requiring a user with an orthodox project setup to provide a `version-file` input?

why would they include a node version config file in their project if they didn't want it to be used?

(if for some reason a project had multiple config files – e.g. `.nvmrc` _and_ `.node-version` – or an unorthodox, custom file name then I agree being _able_ to specify `version-file` would be useful; they seem like quite an edge case though)

Reply:
We love the idea of adding logic to search and parse some node version files by default in case both the `node-version` and `node-version-file` inputs are not specified, but for now we have decided to stick with the current approach. We will probably implement improvements in the near future.

@sndrs feel free to create a [feature request](https://github.com/actions/setup-node/issues/new/choose) to track progress. 
Thanks!